### PR TITLE
Feature/disease observer

### PR DESCRIPTION
**Pull request by michaella21**
disease observer for new disease cases/total susceptible person time by each demographic group by year.

### CODE REVIEW

File: src/vivarium_public_health/metrics/disease.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        pop = pop[pop.alive == 'alive']
+        for group, age_group in self.age_bins.iterrows():
+            start, end = age_group.age_group_start, age_group.age_group_end
+            for sex in ['Male', 'Female']:
+                in_age_group = pop[(pop.age >= start) & (pop.age < end) & (pop.sex == sex)]
+                new_disease_cases = in_age_group[in_age_group[f'{self.disease}_event_time'] == event.time]
+                susceptible_pop = in_age_group[in_age_group[f'{self.disease}'] == f'susceptible_to_{self.disease}']
+                self.frame.loc[(group, event.time.year, sex), f'{self.disease}_cases'] += len(new_disease_cases)
+                self.frame.loc[(group, event.time.year, sex), 'susceptible_person_time'] += \
+                    (event.step_size / pd.Timedelta(days=365)) * len(susceptible_pop)
```
Comment (kfwilson) [inline src/vivarium_public_health/metrics/disease.py]: don't we usually do 365.25?

Comment (michaella21) [reply src/vivarium_public_health/metrics/disease.py]: ah, yeah I think so.

**Comment by halfhorst** [comment]
you may want to add a self.name to the component. I have tried to start adding them - i think we want everything to be named eventually, so we can do dependency management stuff

File: src/vivarium_public_health/metrics/disease.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    """ An observer for disease cases and person time susceptible to
+    the specific disease during simulation. This component by default
+    observes the total number of disease cases and total sum of susceptible
+    person time by each demographic group (age and sex) by each simulation
+    year.
+
+    """
+    def __init__(self, disease: str):
+        self.disease = disease
+        self.name = 'disease_observer'
```
Comment (halfhorst) [inline src/vivarium_public_health/metrics/disease.py]: you should add the disease string to the name so it is unique!

**Comment by collijk** [comment]
James to handle merge into old and new branches

**Comment by michaella21** [comment]
well, github says that PR author cannot approve the own pull request :( But it looks good!

**Pull request merged**